src/components/pages/dashboard/panels/map/mapPanel.js

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+
+import { Indicator } from 'components/shared';
+import {
+  Panel,
+  PanelHeader,
+  PanelHeaderLabel,
+  PanelContent,
+  PanelOverlay,
+  PanelError
+} from 'components/pages/dashboard/panel';
+
+import './mapPanel.css';
+
+const atlas = window.atlas;
+const deviceLayer = 'devices-layer';
+
+export class MapPanel extends Component {
+
+  componentDidMount() {
+    if (this.props.azureMapsKey) {
+      const seattlePosition = new atlas.data.Position(-122.3320708, 47.6062);
+      this.map = new atlas.Map('map', {
+        'subscription-key': this.props.azureMapsKey,
+        center: seattlePosition,
+        zoom: 11
+      });
+
+      this.map.addPins([], {
+        name: deviceLayer,
+        cluster: true,
+        icon: 'pin-blue'
+      });
+
+      this.calculatePins(this.props, true);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.calculatePins(nextProps);
+  }
+
+  calculatePins(props, initialCall = false) {
+    const deviceIds = Object.keys(props.devices);
+    if (!initialCall && deviceIds.join() === Object.keys(this.props.devices).join()) return;
+
+    const devicePins = deviceIds
+      .map(key => props.devices[key])
+      .filter(({ properties }) => properties.latitude && properties.longitude)
+      .map(({ id, properties }) =>
+        new atlas.data.Feature(new atlas.data.Point([properties.longitude, properties.latitude]), {
+          name: id,
+          address: properties.location || ''
+        })
+      );
+
+    if (this.map && devicePins.length) {
+      this.map.addPins(devicePins, {
+        name: deviceLayer
+      });
+
+      const lons = [];
+      const lats = [];
+      devicePins.forEach(({ geometry: { coordinates: [ longitude, latitude ] } }) => {
+        lons.push(longitude);
+        lats.push(latitude);
+      });
+
+      const swLon = Math.min.apply(null, lons);
```

Comment:
reset the center based on these too?

Reply:
Nope, these values will be used to reset the map around a bounding box so that all the devices are in frame.